File: Data Structures & Algorithms/is-palindrome/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isAlphanumeric(char){
        return (
            (char>='a' && char<='z') ||
            (char>='A' && char<='Z') ||
            (char>='0' && char<='9')
        )

    }
    isPalindrome(s) {
        if(!s) return true
        let i=0,j=s.length-1
        while (i<j){
            while(i<j & !this.isAlphanumeric(s[i])){
                i++
            }
            while(i<j & !this.isAlphanumeric(s[j])){
                j--
            }
            if(s[i].toLowerCase()!==s[j].toLowerCase()){
                return false
            }
            i++
            j--
        }
        return true
        
    }
}

File: Data Structures & Algorithms/palindrome-partitioning/submission-0.py
class Solution:
    def partition(self, s: str) -> List[List[str]]:
        res, path = [], []

        def dfs(index):
            if index==len(s):
                res.append(path[:])
                return
            
            for i in range(index, len(s)):
                if self.isPalindrome(s, index, i):
                    path.append(s[index:i+1])
                    dfs(i+1)
                    path.pop()
        
        dfs(0)
        return res

    def isPalindrome(self, s, start, end):
        while start<end:
            if s[start]!=s[end]:
                return False
            start+=1
            end-=1
        return True